src/pytest_cov/engine.py

```diff
@@ -12,6 +14,19 @@
 from .compat import StringIO, workeroutput, workerinput
 
 
+_CovResult = collections.namedtuple("_CovResult", "result,config")
+
+
+def _work_around_config_mutation(cov, method, *args, **kwargs):
+    fn = getattr(cov, method)
+    old_config = cov.config
+    try:
+        cov.config = config = copy.copy(old_config)
```

Comment:
To be extra safe, you want deepcopy.

Reply:
Fortunately coverage only mutates self.config.__dict__

Reply:
You don't like this?
```
@contextlib.contextmanager
def save_cov_config(cov):
    old_config = cov.config
    cov.config = copy.copy(old_config)
    try:
        yield
    finally:
        cov.config = old_config
```
Then you can write:
```
with save_cov_config(cov):
    total = self.cov.html_report(ignore_errors=True, directory=self.cov_report['html'])
res = _CovResult(total, cov.config)
```
This is easier to understand and maintain. If you want a wrapper for _CovResult, you can write that too.

Reply:
I prefer a simple function

Reply:
I wish we had a simple one. The callers have to twist themselves, passing the name of a method to call.  The wrapping in _CovResult isn't even hinted at in the name of the function.
My context manager at least only does one thing.

Most of all, I wish we knew who the maintainer of this plugin was.

Reply:
Hey, I maintain it, or at least I try :/


Using a context manger is more orthogonal and extensible as it can wrap multiple cov method calls or whatever.